JsonEncodingException

We are receiving a lot of crashes with the stack trace pasted below.
The weird thing is that we see such reports only from the Play Console's "ANR & crashes" section but not from within our Crashlytics console.

These crashes seems to have started from version 1.4.3.
We're not sure if this error has user visible impact as from our preliminary tests push notifications do arrive to devices.

Hope this helps:

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.installProvider (ActivityThread.java:6683)
  at android.app.ActivityThread.installContentProviders (ActivityThread.java:6225)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6140)
  at android.app.ActivityThread.access$1200 (ActivityThread.java:235)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1768)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:6986)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1445)
Caused by: com.squareup.moshi.JsonEncodingException: 
  at com.squareup.moshi.JsonReader.syntaxError (JsonReader.java:233)
  at com.squareup.moshi.JsonUtf8Reader.checkLenient (JsonUtf8Reader.java:1051)
  at com.squareup.moshi.JsonUtf8Reader.doPeek (JsonUtf8Reader.java:366)
  at com.squareup.moshi.JsonUtf8Reader.peek (JsonUtf8Reader.java:193)
  at com.squareup.moshi.JsonAdapter$2.fromJson (JsonAdapter.java:134)
  at com.squareup.moshi.JsonAdapter.fromJson (JsonAdapter.java:36)
  at com.pusher.pushnotifications.internal.MoshiConverter.from (MoshiConverter.java:14)
  at com.pusher.pushnotifications.internal.TapeJobQueue$noExceptionConverter$1.from (TapeJobQueue.java:22)
  at com.pusher.pushnotifications.internal.TapeJobQueue$noExceptionConverter$1.from (TapeJobQueue.java:19)
  at com.squareup.tape2.FileObjectQueue$QueueFileIterator.next (FileObjectQueue.java:95)
  at kotlin.collections.CollectionsKt___CollectionsKt.filterNotNullTo (CollectionsKt___CollectionsKt.java:730)
  at kotlin.collections.CollectionsKt___CollectionsKt.filterNotNull (CollectionsKt___CollectionsKt.java:723)
  at kotlin.collections.CollectionsKt.filterNotNull (CollectionsKt.java)
  or                     .first (CollectionsKt.java)
  at com.pusher.pushnotifications.internal.TapeJobQueue.asIterable (TapeJobQueue.java:52)
  at com.pusher.pushnotifications.internal.ServerSyncHandler.<init> (ServerSyncHandler.java:39)
  at com.pusher.pushnotifications.internal.ServerSyncHandler.<init> (ServerSyncHandler.java:14)
  at com.pusher.pushnotifications.internal.ServerSyncHandler$Companion.obtain$pushnotifications_release (ServerSyncHandler.java:80)
  at com.pusher.pushnotifications.PushNotificationsInstance$serverSyncHandler$1.invoke (PushNotificationsInstance.java:106)
  at com.pusher.pushnotifications.PushNotificationsInstance$serverSyncHandler$1.invoke (PushNotificationsInstance.java:93)
  at com.pusher.pushnotifications.PushNotificationsInstance.<init> (PushNotificationsInstance.java:105)
  at com.pusher.pushnotifications.internal.PushNotificationsInitProvider.onCreate (PushNotificationsInitProvider.java:67)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1940)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1915)
  at android.app.ActivityThread.installProvider (ActivityThread.java:6678)
  at android.app.ActivityThread.installContentProviders (ActivityThread.java:6225)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6140)
  at android.app.ActivityThread.access$1200 (ActivityThread.java:235)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1768)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:6986)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1445)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Fixed
+- Fix issue #85 JsonEncodingException when loading stored pending operations
+
 ## [1.4.3] - 2019-06-07
 
 ### Changed

--- a/pushnotifications/src/main/java/com/pusher/pushnotifications/internal/PersistentJobQueue.kt
+++ b/pushnotifications/src/main/java/com/pusher/pushnotifications/internal/PersistentJobQueue.kt
@@ -2,6 +2,7 @@ package com.pusher.pushnotifications.internal
 
 import com.pusher.pushnotifications.logging.Logger
 import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.JsonEncodingException
 import com.squareup.tape2.ObjectQueue
 import com.squareup.tape2.QueueFile
 import java.io.*
@@ -20,6 +21,9 @@ class TapeJobQueue<T: Serializable>(file: File, converter: ObjectQueue.Converter
     override fun from(source: ByteArray): T? {
       return try {
         converter.from(source)
+      } catch (ex: JsonEncodingException) {
+        log.w("Failed to read object data from tape. Continuing without this data.")
+        null
       } catch (ex: JsonDataException) {
         log.w("Failed to read object data from tape. Continuing without this data.")
         null

